opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -466,11 +465,45 @@ public static void addObservation(Event e, JSONObject jsonObject) {
             } catch (JSONException e1) {
                 Timber.e(e1);
             }
+        } else if (AllConstants.GPS.equals(type)) {
+            createGpsObservation(e, jsonObject, value);
         } else {
             createObservation(e, jsonObject, value);
         }
     }
 
+    private static void createGpsObservation(Event e, JSONObject jsonObject, String value) {
+        createObservation(e, jsonObject, value);
+        if (StringUtils.isNotBlank(value)) {
+            String[] valueArr = value.split(" ");
+            String formSubmissionFieldPrefix = getString(jsonObject, KEY);
+            if (valueArr.length >= 2) {
+                String latitude = valueArr[0];
+                String longitude = valueArr[1];
+                String formSubmissionField = formSubmissionFieldPrefix + "_" + AllConstants.GpsConstants.LATITUDE;
+                addObs(e, formSubmissionField, AllConstants.TEXT, Collections.singletonList(latitude));
+
+                formSubmissionField = formSubmissionFieldPrefix + "_" + AllConstants.GpsConstants.LONGITUDE;
+                addObs(e, formSubmissionField, AllConstants.TEXT, Collections.singletonList(longitude));
```

Comment:
The code here looks duplicated, we can maybe move the common code into a private method.